src/hooks/api/use-public-users-client.ts

```diff
@@ -0,0 +1,37 @@
+import { ECHOSTUDY_API_URL } from '@/helpers/api';
+import { PublicUser } from '@/models/public-user';
+import { useFetchWrapper } from './use-fetch-wrapper';
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+/**
+ * Client targeting endpoints public users
+ */
+export function usePublicUsersClient() {
+  const fetchWrapper = useFetchWrapper(ECHOSTUDY_API_URL);
+
+  return { getPublicUsers };
+
+  // GET: /Users/Names
+  async function getPublicUsernames() {
+    const usernames: string[] = await fetchWrapper.get('/users/names');
+    return usernames;
+  }
+
+  // GET: /Users/Username
+  async function getPublicUsers() {
+    const usernames = await getPublicUsernames();
+    const allUsersRequests = usernames.map((username) => fetchWrapper.get(`/users/${username}`));
```

Comment:
i feel like we should have an endpoint that gives all public users as well as some extra data like their `numDecks` so that would don't have to make API calls to the order of users